src/core/config/merge.js

```diff
@@ -15,25 +15,38 @@ import deepExtend from "deep-extend"
 
 const merge = (target, ...sources) => {
   let domNode = Symbol.for("domNode")
-  const sourcesWithoutDomNode = []
+  let primaryName = null
+  const sourcesWithoutExceptions = []
 
   for (const source of sources) {
+    const sourceWithoutExceptions = { ...source }
+
     if (Object.hasOwn(source, "domNode")) {
       domNode = source.domNode
-      const sourceWithoutDomNode = { ...source }
-      delete sourceWithoutDomNode.domNode
-      sourcesWithoutDomNode.push(sourceWithoutDomNode)
-    } else {
-      sourcesWithoutDomNode.push(source)
+      delete sourceWithoutExceptions.domNode
+    }
+
+    if (source["urls.primaryName"]) {
```

Comment:
Let me know if I should change these to 
```js
Object.hasOwn(source, "urls.primaryName") 
```
and
```js
Object.hasOwn(source, "urls") && Object.hasOwn(source.urls, "primaryName") 
```
for consistency with `domNode`.

Reply:
Yes please, Object.hasOwn make sure we're not dealing with inherited prototype properties.